src/ray/raylet/node_manager.cc

```diff
@@ -539,6 +545,66 @@ ray::Status NodeManager::RegisterGcs() {
         "NodeManager.deadline_timer.print_event_loop_stats");
   }
 
+  if (RayConfig::instance().use_ray_syncer()) {
+    // Register resource manager and scheduler
+    ray_syncer_.Register(
+        syncer::RayComponentId::RESOURCE_MANAGER,
+        this,
```

Comment:
I found this part is a bit hard to comprehend. It seems like for `Resource_Manager` component, the node manager is both reporter & receiver. The GCS resource manager is just a receiver. Does this mean the node manager broadcasts the state to all nodes, which means we will start doing all to all communication? Or is GCS still used as a component to broadcast states to all nodes?

Reply:
For our use case, GCS will do broadcasting.
The protocol is: for a connection A and B, A will send all the knowledge it has except the one B sends to B. B will do the same thing.

Right now, all nodes connect to GCS, so GCS will send the raylet the cluster information except the ones sent from that raylet.

Reply:
Hmm in this case, broadcasting messages are independent of reporter & receiver? Based on what you said, my assumption is the gcs resource manager is also supposed to be receiver and reporter?